_posts/2020-05-30-Metal3-Cluster-Deployment.md

```diff
@@ -0,0 +1,354 @@
+```console
+kubectl describe Cluster ${CLUSTER_NAME} -n metal3
+```
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: Cluster
+metadata:
+  [......]
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - 192.168.0.0/18
+    services:
+      cidrBlocks:
+      - 10.96.0.0/12
+  controlPlaneEndpoint:
+    host: 192.168.111.249
+    port: 6443
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+    kind: KubeadmControlPlane
+    name: eko
+    namespace: metal3
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: Metal3Cluster
+    name: eko
+    namespace: metal3
+status:
+  infrastructureReady: true
+  phase: Provisioned
+```
+  
+<br/>
+<br/>
+
+### Provision Controlplane
+
+This script, located at the path - ${metal3-dev-env}/scripts/v1alphaX/provision_clusters.sh, provisions the controlplane or the master of the cluster. As part of the controlplane provisioning, the generated manifests are for cluster, controlplane and machinedeployment. The **MachineDeployment** creates a **Machine** which picks up a BareMetalHost satisfying its requirements as the controlplane node, and it is then provisioned by the BareMetalOperator. there are other resources created in the process of provisioning, like `KubeadmControlPlane` and `Metal3MachineTemplate` resource which are involved in the provisioning process.
+
+> info "Note"
+> It takes some time for the provisioning of the Controlplane, you can watch the process using some steps shared a bit later
+
+```console
+kubectl get KubeadmControlPlane ${CLUSTER_NAME} -n metal3
+kubectl describe KubeadmControlPlane ${CLUSTER_NAME} -n metal3
+```
+
+```yaml
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+kind: KubeadmControlPlane
+metadata:
+  [....]
+  ownerReferences:
+  - apiVersion: cluster.x-k8s.io/v1alpha3
+    blockOwnerDeletion: true
+    controller: true
+    kind: Cluster
+    name: eko
+    uid: aec0f73b-a068-4992-840d-6330bf943d22
+  resourceVersion: "44555"
+  selfLink: /apis/controlplane.cluster.x-k8s.io/v1alpha3/namespaces/metal3/kubeadmcontrolplanes/eko
+  uid: 99487c75-30f1-4765-b895-0b83b0e5402b
+spec:
+  infrastructureTemplate:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: Metal3MachineTemplate
+    name: eko-controlplane
+    namespace: metal3
+  kubeadmConfigSpec:
+    files:
+    - content: |
+        [....]
+  replicas: 1
+  version: v1.18.0
+status:
+  replicas: 1
+  selector: cluster.x-k8s.io/cluster-name=eko,cluster.x-k8s.io/control-plane=
+  unavailableReplicas: 1
+  updatedReplicas: 1
+```
+
+```console
+kubectl get Metal3MachineTemplate ${CLUSTER_NAME}-controlplane -n metal3
+```
+
+To track the progress of provisioning, you can try the following:
+
+```console 
+kubectl get bmh -n metal3 -w
+```
+> bmh=BareMetalHosts
+> You should see all the nodes that were created at the time of metal3 deployment, along with their current status as the provisioning progresses
```

Comment:
I think you don't want that line in a quoted section, right? (starting with `>`)... at the very least you probably want a blank line before it, to prevent it to get joined with the `bmh` clarification